auth/validator.go

```diff
@@ -90,15 +92,25 @@ func (m *Middleware) validateClaims(t Token, ks *oidcclient.OIDCTenant) error {
 	return nil
 }
 
-func (m *Middleware) getOIDCTenant(tokenIssuer string) (*oidcclient.OIDCTenant, error) {
-	issURI, err := m.verifyIssuer(tokenIssuer)
+// getOIDCTenant returns an OIDC Tenant with discovered .well-known/openid-configuration.
+//
+// issuer is the iss (or ias_iss, if set) of the incoming token
+//
+// customIssuer is the iss claim of the incoming token
+func (m *Middleware) getOIDCTenant(issuer, customIssuer string) (*oidcclient.OIDCTenant, error) {
+	issURI, err := m.verifyIssuer(issuer)
 	if err != nil {
 		return nil, err
 	}
 
-	oidcTenant, exp, found := m.oidcTenants.GetWithExpiration(tokenIssuer)
-	if !found || time.Now().After(exp) {
-		newKeySet, err, _ := m.sf.Do(tokenIssuer, func() (i interface{}, err error) {
+	if customIssuer == "" {
```

Comment:
I would introduce  `tokenIssuer` variable which is either `customIssuer` or `issuer` if there is `customIssuer`  as fallback.

Reply:
I need both.

Reply:
ok, but in case customIssuer is not set, "issuer" is not a custom issuer... anyway

Reply:
Yes I get your point.. However what is the different between the Issuer and the TokenIssuer? 
Here I need the Issuer with the sap domain and actual Issuer which is stored in the `iss` domain.
The getter methods we have now are not optimal to solve this requirement. I could name the parameters `ias_iss` and `iss` as they are in the token, but then I would have to call the `Issuer()` method for the parameter `ias_iss` (instead of `iss`). And for `iss` I'd have to check manually if `CustomIssuer()` returns a value, use this one for the `iss` parameter and if not use value of the `Issuer()` method here as well.

So for internal usage the logic behind `Issuer()` and `CustomIssuer()` is not optimal.

Reply:
yep.